cwe_checker_rs/src/analysis/backward_interprocedural_fixpoint/mod.rs

```diff
@@ -1,11 +1,24 @@
+//! Creating and computing backward interprocedural fixpoint problems.
+//!
+//! # General notes
+//!
+//! This module supports computation of fixpoint problems on the control flow graphs generated by the `graph` module.
+//!
+//!
+//! To compute a generalized fixpoint problem,
+//! first construct a context object implementing the `Context`trait.
+//! Use it to construct a `Computation` object.
+//! The `Computation` object provides the necessary methods for the actual fixpoint computation.
+
 use super::fixpoint::Context as GeneralFPContext;
 use super::graph::*;
 use super::interprocedural_fixpoint_generic::*;
 use crate::intermediate_representation::*;
-use fnv::FnvHashMap;
-use petgraph::graph::{EdgeIndex, NodeIndex};
+use petgraph::graph::EdgeIndex;
 use std::marker::PhantomData;
 
+pub mod mock_context;
```

Comment:
The mock context is only used for unit tests, so we should only compile it when executing tests by adding `#[cfg(test)]`.